### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by jahsu-MSFT**
https://github.com/microsoft/azure-pipelines-tasks/issues/11835

### CODE REVIEW

File: src/Misc/dotnet-install.ps1 [hunk 1]
```diff
@@ -167,7 +167,7 @@ function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
         { $_ -eq "x86" } { return "x86" }
         { $_ -eq "arm" } { return "arm" }
         { $_ -eq "arm64" } { return "arm64" }
-        default { throw "Architecture not supported. If you think this is a bug, report it at https://github.com/dotnet/cli/issues" }
+        default { throw "Architecture not supported. If you think this is a bug, report it at https://github.com/dotnet/sdk/issues" }
```
Comment (fadnavistanmay) [inline src/Misc/dotnet-install.ps1]: Is this file change required for the issue?

File: src/Misc/dotnet-install.sh [hunk 2]
```diff
@@ -144,7 +144,7 @@ get_linux_platform_name() {
     else
         if [ -e /etc/os-release ]; then
             . /etc/os-release
-            echo "$ID.$VERSION_ID"
+            echo "$ID${VERSION_ID:+.${VERSION_ID}}"
```
Comment (fadnavistanmay) [inline src/Misc/dotnet-install.sh]: Same here.

File: src/Misc/dotnet-install.ps1 (hunk 1 above)
Comment (fadnavistanmay) [reply src/Misc/dotnet-install.ps1]: Please include this in the description as well.

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 3]
```diff
@@ -270,7 +276,7 @@ private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext conte
         {
             if(String.IsNullOrWhiteSpace(pipelineDefinition)) 
             {
-                throw new InvalidOperationException("Pipeline definition cannot be null or empty");
+                throw new InvalidOperationException(StringUtil.Loc("CannotBeNullOrEmpty", "Pipeline Dedinition"));
```
Comment (fadnavistanmay) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: Definition*

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 4]
```diff
@@ -194,15 +196,19 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
...
                     {
                         pipelineId = await this.GetPipelineIdAsync(context, pipelineDefinition, pipelineVersionToDownload, projectId.ToString(), tagsInput, resultFilter, null, cancellationToken: token);
                     }
                     else if (pipelineVersionToDownload == pipelineVersionToDownloadSpecific)
                     {
-                        pipelineId = Int32.Parse(userSpecifiedpipelineId);
+                        bool isPipelineIdNum = Int32.TryParse(userSpecifiedRunId, out pipelineId);
+                        if(!isPipelineIdNum)
+                        {
+                            throw new ArgumentException("RunId/PipelineId is not a valid number.");
```
Comment (fadnavistanmay) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: A better o/p might be - Pipelineid is not valid - similar to "BuildIdIsNotValid": "Build Id is not valid: {0}",

File: src/Misc/dotnet-install.ps1 (hunk 1 above)
Comment (jahsu-MSFT) [reply src/Misc/dotnet-install.ps1]: This file is from outside of this repo and just we needed to download it to the agent repo

File: src/Misc/dotnet-install.sh (hunk 2 above)
Comment (jahsu-MSFT) [reply src/Misc/dotnet-install.sh]: This file is from outside of this repo and just we needed to download it to the agent repo

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs (hunk 4 above)
Comment (jahsu-MSFT) [reply src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: Sounds good.

**Pull request merged**